algorithms/kernel/pca/oneapi/pca_dense_correlation_online_kernel_ucapi_impl.i

```diff
@@ -0,0 +1,146 @@
+#include "pca_dense_correlation_online_kernel_ucapi.h"
+
+#include <iostream>
+
+using namespace daal::services;
+using namespace daal::internal;
+using namespace daal::oneapi::internal;
+using namespace daal::data_management;
+
+namespace daal
+{
+namespace algorithms
+{
+namespace pca
+{
+namespace internal
+{
+template <typename algorithmFPType>
+PCACorrelationKernelOnlineUCAPI<algorithmFPType>::PCACorrelationKernelOnlineUCAPI(const PCACorrelationBaseIfacePtr & host_impl)
+{
+    _host_impl = host_impl;
+}
+
+template <typename algorithmFPType>
+Status PCACorrelationKernelOnlineUCAPI<algorithmFPType>::compute(const data_management::NumericTablePtr & pData,
+                                                                PartialResult<correlationDense> * partialResult,
+                                                                const OnlineParameter<algorithmFPType, correlationDense> * parameter)
+{
+    DAAL_ITTNOTIFY_SCOPED_TASK(compute);
+    parameter->covariance->input.set(covariance::data, pData);
+    parameter->covariance->parameter.outputMatrixType = covariance::correlationMatrix;
+
+    services::Status s;
+    {
+        DAAL_ITTNOTIFY_SCOPED_TASK(compute.covariance);
+        s = parameter->covariance->computeNoThrow();
+    }
+    if (s) s = copyCovarianceResultToPartialResult(parameter->covariance->getPartialResult().get(), partialResult);
+
+    return s;
+}
+
+template <typename algorithmFPType>
+services::Status PCACorrelationKernelOnlineUCAPI<algorithmFPType>::copyCovarianceResultToPartialResult(
+    covariance::PartialResult * covariancePres, PartialResult<correlationDense> * partialResult)
+{
+    DAAL_ITTNOTIFY_SCOPED_TASK(compute.copy.to.partial.result);
+    services::Status s = copyIfNeeded(covariancePres->get(covariance::sum).get(), partialResult->get(sumCorrelation).get());
+    s |= copyIfNeeded(covariancePres->get(covariance::nObservations).get(), partialResult->get(nObservationsCorrelation).get());
+    s |= copyIfNeeded(covariancePres->get(covariance::crossProduct).get(), partialResult->get(crossProductCorrelation).get());
+    return s;
+}
+
+template <typename algorithmFPType>
+services::Status PCACorrelationKernelOnlineUCAPI<algorithmFPType>::copyIfNeeded(data_management::NumericTable * src,
+                                                                                  data_management::NumericTable * dst)
+{
+    if (src == dst) return services::Status();
+
+    services::Status status;
+
+    BlockDescriptor<algorithmFPType> srcBlock;
+    BlockDescriptor<algorithmFPType> dstBlock;
+
+    {
+        status |= src->getBlockOfRows(0, src->getNumberOfRows(), readOnly, srcBlock);
+        DAAL_CHECK_STATUS_VAR(status);
+
+        status |= dst->getBlockOfRows(0, dst->getNumberOfRows(), writeOnly, dstBlock);
+        DAAL_CHECK_STATUS_VAR(status);
+    }
+
+    const size_t nRows = dst->getNumberOfRows();
+    const size_t nCols = dst->getNumberOfColumns();
+    const size_t nDataElements = nRows * nCols;
```

Comment:
The shadowed assumption that `src` and `dst` have the same size. I'd add a debug assertion at least.